queue/queue.mbt

```diff
@@ -0,0 +1,351 @@
+struct Cell[T] {
+  content : T
+  mut next : Option[Cell[T]]
+}
+
+pub fn op_equal[T : Eq](self : Cell[T], other : Cell[T]) -> Bool {
+  self.content == other.content
+}
+
+pub struct Queue[T] {
+  mut length : Int
```

Comment:
note for type declarations, the default is `abstract` type, which is exactly what you need.
the `pub` means export everything